### PR TITLE
ContributionSummary

### PR DESCRIPTION
**Pull request by gza007**

### CODE REVIEW

File: src/components/contribution-summary.jsx [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
-export default ContributionSummary;
+  componentDidMount() {
+    fetch(`https://mcr-codes-cohorts.herokuapp.com/users/${this.props.username}`)
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ contributions: data.contributions });
+        return data;
+      })
+      .then(data => {
+        this.setState({ pushEvents: data.events.filter(event => event.type === 'PushEvent') });
```
Comment (ersel) [inline src/components/contribution-summary.jsx]: `pushEvents` and `pullEvents` arrays can be constructed in a single then block.

```
.then(data => {
  const pushEvents = data.events.filter(event => event.type === 'PushEvent');
  const pullEvents = data.events.filter(event => event.type === 'PullEvent');
  this.setState({ pushEvents,  pullEvents });
})
```

**Comment by gza007** [comment]
Changes completed.

File: src/components/contribution-summary.jsx [hunk 2]
```diff
@@ -1,5 +1,53 @@
...
+        const pushEvents = data.events.filter(event => event.type === 'PushEvent');
+        const pullRequests = data.events.filter(event => event.type === 'PullRequestEvent');
+        this.setState({ pushEvents, pullRequests });
+      });
+  }
+
+
+  render() {
+    const { contributions } = this.state;
+    return this.state.contributions ? (
```
Comment (ersel) [inline src/components/contribution-summary.jsx]: just use `contributions`

Comment (ersel) [reply src/components/contribution-summary.jsx]: destructure other state fields `pushEvents` and `pullRequests`

**Pull request merged**